sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_models.hpp

```diff
@@ -0,0 +1,772 @@
+    {
+      if (value.empty())
+      {
+        throw std::invalid_argument("The value for the curve name can not be empty");
+      }
+      m_value = std::move(value);
+    }
+
+    /**
+     * @brief Construct a default key curve.
+     *
+     */
+    CertificatePolicyAction() = default;
+
+    /**
+     * @brief Enables using the equal operator for key curve.
+     *
+     * @param other A key curve to be compared.
+     */
+    bool operator==(const CertificatePolicyAction& other) const noexcept
+    {
+      return m_value == other.m_value;
+    }
+
+    /**
+     * @brief Get the string value of the key curve.
+     *
+     */
+    std::string const& ToString() const { return m_value; }
+
+    /**
+     * @brief Get the NIST P-256 elliptic curve, AKA SECG curve SECP256R1.
+     *
+     * @remark For more information, see
+     *  <a href="https://docs.microsoft.com/azure/key-vault/keys/about-keys#curve-types">Curve
+     * types</a>.
+     *
+     */
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction
+        DigitalSignature;
+
+    /**
+     * @brief Get the SECG SECP256K1 elliptic curve.
+     *
+     * @remark For more information, see
+     * <a href="https://docs.microsoft.com/azure/key-vault/keys/about-keys#curve-types">Curve
+     * types</a>.
+     *
+     */
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction NonRepudiation;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction
+        KeyEncipherment;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction
+        DataEncipherment;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction KeyAgreement;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction KeyCertSign;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction CrlSign;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction EncipherOnly;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction DecipherOnly;
+  };
+
+  /**
+   * @brief An action to be executed at a prescribed time in a certificates lifecycle.
+   *
+   */
+  struct LifetimeAction final
+  {
+    /**
+     * @brief Get the CertificatePolicyAction to be performed.
+     *
+     */
+    CertificatePolicyAction Action;
+
+    /**
+     * @brief Get the action should be performed the specified number of days before the certificate
+     * will expire.
+     *
+     */
+    Azure::Nullable<int32_t> DaysBeforeExpiry;
+
+    /**
+     * @brief Get the action should be performed when the certificate reaches the specified
+     * percentage of its lifetime. Valid values include 1-99.
+     *
+     */
+    Azure::Nullable<int32_t> LifetimePercentage;
+  };
+
+  /**
+   * @brief Properties of a CertificateIssuer.
+   *
+   */
+  struct IssuerParameters final
```

Comment:
We don't expose this in .NET or Java, and hide it in other languages. Shouldn't this go under `_details` or something? It should be used only for de/serialization.

Reply:
@vhvb1989, you can get away with it:
```
struct SubjectAlternativeNames { int x; };

struct CertificatePolicy
{
    struct SubjectAlternativeNames SubjectAlternativeNames;
};
```

Reply:
Move this into _details since we do not expect customers to use it.

Reply:
> we do not expect customers to use it.

How come?